e2e/internal/kubeclient/deploy.go

```diff
@@ -225,6 +234,31 @@ func (c *Kubeclient) checkIfRunning(ctx context.Context, name string, namespace
 	return true, nil
 }
 
+// Checks whether the FailedCreatePodSandBox Event occurred which indicates that the SetPolicy request is rejected and the Kata Shim fails to start the Pod sandbox.
+func (c *Kubeclient) checkIfStartingBlocked(name string, namespace string, resource ResourceWaiter, evt watch.Event, startingPoint time.Time) (bool, error) {
```

Comment:
Consider renaming to `isStartingBlocked`. Also, please check the `static-checks` job results on your PR, it should have complained about the docstring format (must start with the function name).